src/modules/view-licences/controller.js

```diff
@@ -191,10 +192,65 @@ const getLicence = async (request, h) => {
   return h.view('nunjucks/view-licences/licence.njk', view, { layout: false });
 };
 
+const validateMessageAccess = (request, companyId) => {
+  const isAdmin = request.permissions.hasPermission('admin.defra');
+  const entityCompanies = request.entityRoles.map(role => role.company_entity_id);
+
+  return isAdmin || entityCompanies.includes(companyId);
+};
+
+const getAddressParts = notification => {
+  return [
+    'addressLine1',
+    'addressLine2',
+    'addressLine3',
+    'addressLine4',
+    'addressLine5',
+    'postcode'
+  ].reduce((acc, part) => {
+    const addressPart = trim(notification.address[part]);
+    return addressPart ? [...acc, addressPart] : acc;
+  }, []);
+};
+
+const validateLicenceCommunicationResponses = (request, licence) => {
+  if (!licence) {
+    return Boom.notFound('Document not associated with communication');
+  }
+
+  if (!validateMessageAccess(request, licence.companyEntityId)) {
+    return Boom.forbidden();
+  }
+};
+
+const getLicenceCommunication = async (request, h) => {
+  const { communicationId, documentId } = request.params;
+  const response = await communicationsConnector.getCommunication(communicationId);
+
+  const licence = response.data.licenceDocuments.find(doc => doc.documentId === documentId);
+
+  const validationError = validateLicenceCommunicationResponses(request, licence);
+  if (validationError) return validationError;
```

Comment:
I'd prefer to stick with throw here for now, as that's the pattern used elsewhere in the service